.github/workflows/johnnytest4.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+output_file="CMake_Error_Log.txt"
+
+# Clean the output file first
+echo "" > $output_file
+
+# specify the directory
+directory="./.github/workflows"
+
+# loop over each file in directory
+for input_file in $directory/*.log; do
+
+  echo "File: $input_file" >> $output_file
+  echo "" >> $output_file
+  echo "Error Message:" >> $output_file
+  echo "" >> $output_file
+
+  awk '/CMake Error/{flag=1}/^$/{flag=0}flag' $input_file >> $output_file
+
+  # Load error message into a variable
+  error_message=$(awk '/CMake Error/{flag=1}/^$/{flag=0}flag' $input_file)
+  
+  if [[ -z "$error_message" ]]; then
+    echo "No CMake errors detected." >> $output_file
+    continue
+  fi
+
```

Comment:
Moreover I may put the echo $error_message >> $output_file into an ELSE branchlike this:
```
if [[ -z "$error_message" ]]; then
    echo "No CMake errors detected." >> $output_file
    continue
else
    echo $error_message >> $output_file
fi
```